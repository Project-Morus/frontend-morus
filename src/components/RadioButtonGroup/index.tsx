import RadioButton from "../RadioButton";
import { FieldsetSC, WrapperSC } from "./styles";
import { IOptionGroupProps, IOptionProps } from "./types";

const RadioButtonGroup = ({ label = "", options, onChange, required, name }: IOptionGroupProps) => {
  const renderOptions = () => {
    return options.map(({ label, disabled }: IOptionProps, index) => {
      const shortenedOptionLabel = label.replace(/\s+/g, "");
      const optionId = `radio-option-${shortenedOptionLabel}`;

      return (
        <RadioButton
          key={optionId}
          value={label}
          label={label}
          id={optionId}
          name={name}
          required={required}
          disabled={disabled}
          defaultChecked={index === 0}
          onChange={onChange}
        />
      );
    });
  };

  return (
    <FieldsetSC>
      <legend>{label}</legend>
      <WrapperSC>{renderOptions()}</WrapperSC>
    </FieldsetSC>
  );
};

export default RadioButtonGroup;
