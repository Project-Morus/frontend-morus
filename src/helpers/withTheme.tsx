import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

