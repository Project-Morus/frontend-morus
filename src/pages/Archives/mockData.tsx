import { ITableHeaderProps } from "../../components/DataDisplay/Table/types";
import Icons from "./parts/Icons";


export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Tamanho" },
  { colName: "Nome" },
  { colName: "Data da postagem" },
  { colName: "Ações" },
];

export const ROW_NAME_CELLS = [
  { size: "35MB", name: "Regimento Interno do Condomínio Ilha de Capri", created_at: "11/05/2023", actions: <Icons /> },
  { size: "15MB", name: "Convenção do condomínio Ilha de Capri", created_at: "12/04/2023", actions: <Icons /> },
  { size: "2MB", name: "Edital de Convocação para Assembleia Geral Ordinária", created_at: "10/04/2023", actions: <Icons /> },
  { size: "7MB", name: "Alvará de construção da prefeitura de Vila Velha-ES", created_at: "12/02/2022", actions: <Icons /> },
  { size: "101MB", name: "Plantas e Projetos", created_at: "19/01/2022", actions: <Icons /> },
];
