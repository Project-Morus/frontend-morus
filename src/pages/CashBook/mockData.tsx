import Icon from "../../components/Icon";
import { ITableHeaderProps } from "../../components/Table/types";

const IconsSC = () => {
  return (
    <div style={{ display: "flex", gap: "0.8rem" }}>
      <Icon name="ph-pencil-simple-line" size="20" />
      <Icon name="ph-trash" color="#CE323B" size="20" />
    </div>
  );
};

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Torre" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Data" },
  { colName: "Categoria" },
  { colName: "Ações" },
];

export const ROW_NAME_CELLS = [
  {
    tower: "Todos",
    description: "Salário do porteiro",
    value: "-R$3.250,50",
    date: "05/02/2024",
    category: "Funcionário",
    actions: <IconsSC />,
  },
  {
    tower: "Norte",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Sul",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Sul",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Norte",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
];
