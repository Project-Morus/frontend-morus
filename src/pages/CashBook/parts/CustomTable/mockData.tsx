import { ITableHeaderProps } from "../../types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Torre" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Data" },
  { colName: "Categoria" },
  { colName: "Ações" },
];
