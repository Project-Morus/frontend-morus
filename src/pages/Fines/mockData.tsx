import { ITableHeaderProps } from "../../components/DataDisplay/Table/types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Nome" },
  { colName: "Motivo" },
  { colName: "Valor" },
  { colName: "Juros" },
  { colName: "Data" },
  { colName: "Vencimento" },
  { colName: "Status" },
  { colName: "Ações" },
];
