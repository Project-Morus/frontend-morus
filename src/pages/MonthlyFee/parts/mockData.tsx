import { ITableHeaderProps } from "../../../components/DataDisplay/Table/types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Nome" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Vencimento" },
  { colName: "Status" },
  { colName: "Ações" },
];
