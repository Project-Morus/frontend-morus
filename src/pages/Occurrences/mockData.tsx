import { ITableHeaderProps } from "../../components/DataDisplay/Table/types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Motivo" },
  { colName: "Descrição" },
  { colName: "Data da Ocorrência" },
];
