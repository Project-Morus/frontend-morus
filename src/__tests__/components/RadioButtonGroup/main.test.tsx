import { fireEvent, render, screen } from "@testing-library/react";
import RadioButtonGroup from "../../../components/RadioButtonGroup";
import { withTheme } from "../../../helpers/withTheme";
import { RADIO_BUTTON_GROUP_DATA } from "./mockData";

const RadioButtonGroupWithTheme = withTheme(RadioButtonGroup);

const mockOnChange = jest.fn()

const name = 'drink-types'

describe("RadioButtonGroup Component", () => {
  test("Verify if component is rendering", () => {
    render(<RadioButtonGroupWithTheme options={RADIO_BUTTON_GROUP_DATA} label="Name" onChange={mockOnChange} name={name} />);
    expect(screen.getByText(RADIO_BUTTON_GROUP_DATA[0].label)).toBeVisible()
  });

  test("Verify if label is rendering correctly", () => {
    render(<RadioButtonGroupWithTheme label="Name" options={RADIO_BUTTON_GROUP_DATA} onChange={mockOnChange} name={name} />);
    const labelElement = screen.getByText("Name");

    expect(labelElement).toBeVisible();
  });

  test('Correctly invokes onChange when selecting an option', () => {
    render(<RadioButtonGroupWithTheme label="Name" options={RADIO_BUTTON_GROUP_DATA} onChange={mockOnChange} name={name} />);

    fireEvent.click(screen.getByText(RADIO_BUTTON_GROUP_DATA[1].label));
    expect(mockOnChange).toHaveBeenCalledTimes(1);

    fireEvent.click(screen.getByText(RADIO_BUTTON_GROUP_DATA[0].label));
    expect(mockOnChange).toHaveBeenCalledTimes(2);
  });

  test("Snapshot RadioButtonGroup", () => {
    const { asFragment } = render(<RadioButtonGroupWithTheme options={RADIO_BUTTON_GROUP_DATA} label="Name" onChange={mockOnChange} name={name} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
