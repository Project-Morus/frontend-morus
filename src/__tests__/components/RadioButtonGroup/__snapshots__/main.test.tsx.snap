// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup Component Snapshot RadioButtonGroup 1`] = `
<DocumentFragment>
  .c3 {
  width: 20px;
  height: 20px;
  appearance: none;
  border-radius: 50%;
  border: 2px solid #023B74;
  cursor: pointer;
  transition: border-color 0.2s;
}

.c3:checked {
  border-width: 5px;
}

.c3:not(:checked):hover {
  border-color: #0353A4;
}

.c3:disabled {
  opacity: 0.5;
  cursor: default;
}

.c2 {
  width: 100%;
  display: flex;
  gap: 0.5rem;
  border: 1px solid #BABABA;
  border-radius: 5px;
  background-color: #FFF;
  padding: 1rem;
}

.c5 {
  width: 100%;
  display: flex;
  gap: 0.5rem;
  border: 1px solid #BABABA;
  border-radius: 5px;
  background-color: #FFF;
  padding: 1rem;
  opacity: 0.5;
}

.c4 {
  color: #4E4E4E;
}

.c6 {
  color: #4E4E4E;
  opacity: 0.5;
}

.c0 {
  border: none;
}

.c1 {
  display: flex;
  gap: 1rem;
  margin-top: 0.75rem;
}

<fieldset
    class="c0"
  >
    <legend>
      Name
    </legend>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <input
          checked=""
          class="c3"
          id="radio-option-Coffee"
          name="drink-types"
          type="radio"
          value="Coffee"
        />
        <label
          class="c4"
          for="radio-option-Coffee"
        >
          Coffee
        </label>
      </div>
      <div
        class="c2"
      >
        <input
          class="c3"
          id="radio-option-Tea"
          name="drink-types"
          type="radio"
          value="Tea"
        />
        <label
          class="c4"
          for="radio-option-Tea"
        >
          Tea
        </label>
      </div>
      <div
        class="c5"
      >
        <input
          class="c3"
          disabled=""
          id="radio-option-Cerveja"
          name="drink-types"
          type="radio"
          value="Cerveja"
        />
        <label
          class="c6"
          for="radio-option-Cerveja"
        >
          Cerveja
        </label>
      </div>
    </div>
  </fieldset>
</DocumentFragment>
`;
